drivebrain_app/DriveBrainApp.cpp

```diff
@@ -0,0 +1,239 @@
+        // feedback
+        state_estimator_->set_previous_control_output(out_struct);
+        // output
+        // if(state_and_validity.second)
+        // {
+        out_msg->set_prev_mcu_recv_millis(out_struct.mcu_recv_millis);
+
+        if(temp_desired_torques.res_torque_lim_nm.FL < 0) {
+            out_msg->mutable_desired_rpms()->set_fl(0);
+        } else {
+            out_msg->mutable_desired_rpms()->set_fl(out_struct.desired_rpms.FL);
+        }
+
+        if(temp_desired_torques.res_torque_lim_nm.FR < 0) {
+            out_msg->mutable_desired_rpms()->set_fr(0);
+        } else {
+            out_msg->mutable_desired_rpms()->set_fr(out_struct.desired_rpms.FR);
+        }
+
+        if(temp_desired_torques.res_torque_lim_nm.RL < 0) {
+            out_msg->mutable_desired_rpms()->set_rl(0);
+        } else {
+            out_msg->mutable_desired_rpms()->set_rl(out_struct.desired_rpms.RL);
+        }
+
+        if(temp_desired_torques.res_torque_lim_nm.RR < 0) {
+            out_msg->mutable_desired_rpms()->set_rr(0);
+        } else {
+            out_msg->mutable_desired_rpms()->set_rr(out_struct.desired_rpms.RR);
+        }
+
+        out_msg->mutable_torque_limit_nm()->set_fl(::abs(temp_desired_torques.res_torque_lim_nm.FL));
+        out_msg->mutable_torque_limit_nm()->set_fr(::abs(temp_desired_torques.res_torque_lim_nm.FR));
+        out_msg->mutable_torque_limit_nm()->set_rl(::abs(temp_desired_torques.res_torque_lim_nm.RL));
+        out_msg->mutable_torque_limit_nm()->set_rr(::abs(temp_desired_torques.res_torque_lim_nm.RR));
+
+        {
+            std::unique_lock lk(eth_tx_queue_.mtx);
+            eth_tx_queue_.deque.push_back(out_msg);
+            eth_tx_queue_.cv.notify_all();
+        }
+
+        auto end_time = std::chrono::high_resolution_clock::now();
+        auto elapsed = std::chrono::duration_cast<std::chrono::microseconds>(end_time - start_time);
+        
+        if (loop_chrono_time > elapsed) {
+            std::this_thread::sleep_for(loop_chrono_time - elapsed);
+        }
+    }
+}
+
+void DriveBrainApp::startThreads() {
+    db_service_thread_ = std::make_unique<std::thread>([this]() {
+        spdlog::info("started db service thread");
+        try {
+            while (!stop_signal_.load()) {
+                // Run the io_context as long as stop_signal is false
+                db_service_->run_server(); // Run at least one handler, or return immediately if none
+            }
+        } catch (const std::exception& e) {
+            spdlog::error("Error in drivebrain service thread: {}", e.what());
+        }
+    });
+
+    // what we will do here is have a temporary super-loop.
+    // in this thread we will block on having anything in the rx queue, everything by default goes into the foxglove server (TODO)
+    // if we receive the pedals message, we step the controller and get its output to put intot he tx queue
+    
+    io_context_thread_ = std::make_unique<std::thread>([this]() {
```

Comment:
see comment about member initializer lists for the thread creation in the constructor of the class